core/commands/repo.go

```diff
@@ -298,63 +291,35 @@ var repoVerifyCmd = &oldcmds.Command{
 		for k := range keys {
 			_, err := bs.Get(k)
 			if err != nil {
-				select {
-				case out <- &VerifyProgress{
+				if res.Emit(&VerifyProgress{
 					Msg: fmt.Sprintf("block %s was corrupt (%s)", k, err),
-				}:
-				case <-req.Context().Done():
+				}) != nil {
 					return
 				}
 				fails++
 			}
 			i++
-			select {
-			case out <- &VerifyProgress{Progress: i}:
-			case <-req.Context().Done():
+			if res.Emit(&VerifyProgress{Progress: i}) != nil {
 				return
 			}
 		}
 
 		if fails == 0 {
-			select {
-			case out <- &VerifyProgress{Msg: "verify complete, all blocks validated."}:
-			case <-req.Context().Done():
-				return
-			}
+			res.Emit(&VerifyProgress{Msg: "verify complete, all blocks validated."})
 		} else {
 			res.SetError(fmt.Errorf("verify complete, some blocks were corrupt"), cmdkit.ErrNormal)
 		}
 	},
 	Type: &VerifyProgress{},
-	Marshalers: oldcmds.MarshalerMap{
-		oldcmds.Text: func(res oldcmds.Response) (io.Reader, error) {
-			v, err := unwrapOutput(res.Output())
-			if err != nil {
-				return nil, err
-			}
-
-			obj, ok := v.(*VerifyProgress)
-			if !ok {
-				return nil, e.TypeErr(obj, v)
-			}
-
-			buf := new(bytes.Buffer)
-			if strings.Contains(obj.Msg, "was corrupt") {
-				fmt.Fprintln(os.Stdout, obj.Msg)
```

Comment:
I removed this distinction. Was it really necessary? (also, I assume this was *supposed* to print to stderr...)

Reply:
It was probably to be able to pipe just the corrupted blocks out into other command.

Reply:
I'd guess this was mean to write to stderr as the normal output is written to stdout.